examples/sort-axis.rs

```diff
@@ -102,21 +114,27 @@ where
         assert_eq!(axis_len, perm.indices.len());
         debug_assert!(perm.correct());
 
+        if self.is_empty() {
+            return self;
+        }
+
         let mut result = Array::uninit(self.dim());
 
         unsafe {
             // logically move ownership of all elements from self into result
             // the result realizes this ownership at .assume_init() further down
             let mut moved_elements = 0;
-            for i in 0..axis_len {
-                let perm_i = perm.indices[i];
-                Zip::from(result.index_axis_mut(axis, perm_i))
-                    .and(self.index_axis(axis, i))
-                    .for_each(|to, from| {
-                        copy_nonoverlapping(from, to.as_mut_ptr(), 1);
-                        moved_elements += 1;
-                    });
-            }
+            Zip::from(&perm.indices)
```

Comment:
tip, we can use slices just fine in `Zip`.

Reply:
Thanks for the tip, I haven't used `Zip` much but should start.  I noticed this change being beyond the index swap.